crates/iceberg-catalog/src/service/storage/s3.rs

```diff
@@ -608,13 +516,44 @@ impl S3Profile {
     }
 }
 
-fn is_valid_bucket_name(bucket: &str) -> Result<(), ValidationError> {
+pub(super) fn get_file_io_from_table_config(
+    config: &TableConfig,
+) -> Result<iceberg::io::FileIO, FileIoError> {
+    let mut builder = iceberg::io::FileIOBuilder::new("s3");
+
+    for key in [
+        iceberg::io::S3_REGION,
+        iceberg::io::S3_ENDPOINT,
+        iceberg::io::S3_ACCESS_KEY_ID,
+        iceberg::io::S3_SECRET_ACCESS_KEY,
+        iceberg::io::S3_SESSION_TOKEN,
+    ] {
+        if let Some(value) = config.get_custom_prop(key) {
```

Comment:
get_custom_prop removes type-check on read, this means we defer failure to opendal / reqsign

Reply:
This is already typechecked from insertion - I switched to our own types though.